islandora.module

```diff
@@ -316,6 +316,51 @@ function islandora_preprocess_node(&$variables) {
   }
 }
 
+/**
+ * Implements hook_form_alter().
+ */
+function islandora_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $media_add_forms = ['media_audio_add_form', 'media_document_add_form',
+    'media_extracted_text_add_form', 'media_file_add_form', 'media_image_add_form',
+    'media_fits_technical_metadata_add_form', 'media_video_add_form',
+  ];
+
+  if (in_array($form['#form_id'], $media_add_forms)) {
+    $params = \Drupal::request()->query->all();
+    if (count($params) > 0 && array_key_exists('edit', $params)) {
+      $media_of_nid = $params['edit']['field_media_of']['widget'][0]['target_id'];
+      $node = \Drupal::entityTypeManager()->getStorage('node')->load($media_of_nid);
+      if ($node) {
+        $form['name']['widget'][0]['value']['#default_value'] = $node->getTitle();
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_field_widget_WIDGET_TYPE_form_alter().
+ */
+function islandora_field_widget_image_image_form_alter(&$element, $form_state, $context) {
+  $element['#process'][] = 'islandora_add_default_image_alt_text';
+}
+
+/**
+ * Callback for hook_field_widget_WIDGET_TYPE_form_alter().
+ */
+function islandora_add_default_image_alt_text($element, $form_state, $form) {
+  if ($element['alt']['#access']) {
+    $params = \Drupal::request()->query->all();
+    if (count($params) > 0 && array_key_exists('edit', $params)) {
```

Comment:
Bit of a nit but you could just do an `isset($params['edit'])` or similarly this could be re-written as:

```
 if ($edit_param = \Drupal::request()->query->get('edit')) {
      $media_of_nid = $edit_param['field_media_of']['widget'][0]['target_id'];
      $node = \Drupal::entityTypeManager()->getStorage('node')->load($media_of_nid);
      if ($node) {
        $element['alt']['#default_value'] = $node->getTitle();
      }
    }
```

Not something that would stop me from approving this but just a code style thing.

Reply:
Cool, I've replaced `if (count($params) > 0 && array_key_exists('edit', $params)) {` with the much more concise `if (isset($params['edit'])) {`, but I'll keep the others.

Thanks again for the thorough review!